src/ripple/app/paths/tests/Path_test.cpp

```diff
@@ -1,212 +1,585 @@
-
-        auto master = createAccount("masterpassphrase", KeyType::ed25519);
-
-        std::shared_ptr<Ledger const> LCL;
-        Ledger::pointer ledger;
-        std::tie(LCL, ledger) = createGenesisLedger(100000 * xrp, master);
-
-        // Create accounts
-        auto accounts = createAndFundAccounts(master, { "alice", "bob" },
-            KeyType::ed25519, 10000 * xrp, ledger);
-        auto& alice = accounts["alice"];
-        auto& bob = accounts["bob"];
-        expectNotEquals(alice.pk.humanAccountID(), master.pk.humanAccountID());
-
-        // Set credit limit
-        trust(bob, alice, "USD", 700, ledger);
-
-        // Find path from alice to bob
-        auto alternatives = findPath(ledger, alice, bob, { Currency("USD") },
-            Amount(5, "USD", bob), log.stream());
-        log << "ripplePathFind alternatives: " << alternatives;
+        Env env(*this);
+        env.fund(XRP(10000), "alice", "bob");
+        env.trust(Account("alice")["USD"](700), "bob");
+
+        auto const alternatives = findPath(env.ledger, "alice", "bob",
+            {Account("alice")["USD"](0)}, Account("bob")["USD"](5));
+        Json::Value jv;
+        Json::Reader().parse(R"([{
+		        "paths_canonical" : [],
+		        "paths_computed" : [],
+		        "source_amount" :
+		        {
+			        "currency" : "USD",
+			        "issuer" : "rG1QQv2nh2gr7RCZ1P8YYcBUKCCN633jCn",
+			        "value" : "5"
+		        }
+            }])", jv);
+        expect(jv == alternatives);
+    }
 
-        expectEquals(alternatives.size(), 1);
-        auto alt = alternatives[0u];
-        expectEquals(alt[jss::paths_canonical].size(), 0);
-        expectEquals(alt[jss::paths_computed].size(), 0);
-        auto srcAmount = alt[jss::source_amount];
-        expectEquals(srcAmount[jss::currency], "USD");
-        expectEquals(srcAmount[jss::value], "5");
-        expectEquals(srcAmount[jss::issuer], alice.pk.humanAccountID());
+    void
+    test_payment_auto_path_find()
+    {
+        testcase("payment auto path find");
+        Env env(*this);
+        auto const gw = Account("gateway");
+        auto const USD = gw["USD"];
+        env.fund(XRP(10000), "alice", "bob", gw);
+        env.trust(USD(600), "alice");
+        env.trust(USD(700), "bob");
+        env(pay(gw, "alice", USD(70)));
+        env(pay("alice", "bob", USD(24)));
+        env.require(balance("alice", USD(46)));
+        env.require(balance("bob", USD(24)));
     }
 
     void
-    test_payment_auto_path_find_using_build_path()
+    test_path_find()
     {
-        testcase("payment auto path find (using build_path)");
+        testcase("path find");
+        Env env(*this);
+        auto const gw = Account("gateway");
+        auto const USD = gw["USD"];
+        env.fund(XRP(10000), "alice", "bob", gw);
+        env.trust(USD(600), "alice");
+        env.trust(USD(700), "bob");
+        env(pay(gw, "alice", USD(70)));
+        env(pay(gw, "bob", USD(50)));
+
```

Comment:
I prefer variables over strings for `alice` and `bob` when they are used over multiple lines like this. Like you did with `gw`.